### PR TITLE
Fix appraisals and issue for guest users

### PR DESCRIPTION
**Pull request by harsh-ag**

### CODE REVIEW

File: app/models/spree/order_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+  def update_multi_coupon_codes
+    promotion_ids = all_adjustments.promotion.eligible.map { |a| a.source.promotion_id }
+    promotion = promotions.where(id: promotion_ids).where(multi_coupon: true).last
+    update_used_for_promotion_code(promotion)
+    update_used_for_promotion_code(create_line_item_promotion)
+  end
+
+  private
+
+    def create_line_item_promotion
```
Comment (sawangupta92) [inline app/models/spree/order_decorator.rb]: doesn't seems right to me

File: app/models/spree/order_decorator.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+      promotions.joins(:promotion_actions).where(spree_promotion_actions: { type: 'Spree::Promotion::Actions::CreateLineItems' }).last
```
Comment (sawangupta92) [inline app/models/spree/order_decorator.rb]: why are we only marking it true for Spree::Promotion::Actions::CreateLineItems?

File: app/models/spree/order_decorator.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
+Spree::Order.class_eval do
+
+  state_machine.after_transition to: :complete, do: :update_multi_coupon_codes
+
+  def update_multi_coupon_codes
+    promotion_ids = all_adjustments.promotion.eligible.map { |a| a.source.promotion_id }
+    promotion = promotions.where(id: promotion_ids).where(multi_coupon: true).last
```
Comment (sawangupta92) [inline app/models/spree/order_decorator.rb]: can be replaced with?

promotions.find_by(id: promotion_ids, multi_coupon: true)

File: app/models/spree/promotion_decorator.rb [hunk 4]
```diff
@@ -26,9 +26,9 @@ def multi_coupon_eligible?(promotable)
     def authorized?(promotable)
       promotable_order = promotable.is_a?(Spree::Order) ? promotable : promotable.order
       promotable_user, coupon_code = promotable_order.user, promotable_order.coupon_code
-      already_included_in_order = promotable_order.promotions.include?(self) || promotable_order.all_adjustments.eligible.any? { |adjustment| adjustment.source.promotion == self }
+      already_included_in_order = promotable_order.promotions.include?(self) || promotable_order.all_adjustments.eligible.promotion.any? { |adjustment| adjustment.source.promotion == self }
 
-      if coupon_code.blank? || codes.where("user_id = ? AND lower(code) = ?", promotable_user.id, coupon_code).exists?
+      if coupon_code.blank? || codes.where("user_id = ? AND lower(code) = ?", promotable_user.try(:id), coupon_code).exists?
```
Comment (sawangupta92) [inline app/models/spree/promotion_decorator.rb:31]: was this giving exception?

File: app/models/spree/order_decorator.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
+Spree::Order.class_eval do
+
+  state_machine.after_transition to: :complete, do: :update_multi_coupon_codes
+
+  def update_multi_coupon_codes
```
Comment (sawangupta92) [inline app/models/spree/order_decorator.rb:5]: make sure that this one get tested properly.

**Review by sawangupta92 — CHANGES_REQUESTED** [review]

File: app/models/spree/promotion_decorator.rb (hunk 4 above)
Comment (harsh-ag) [reply app/models/spree/promotion_decorator.rb:31]: this was done for guest users.

**Review by sawangupta92 — APPROVED** [review]

**Pull request merged**